src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -401,15 +401,14 @@ def recover_vault_or_hsm(cmd, client, resource_group_name=None, location=None, v
 
 
 def recover_hsm(cmd, client, hsm_name, resource_group_name, location, no_wait=False):
-    from azure.cli.core._profile import Profile
+    from azure.cli.core._profile import Profile, _TENANT_ID
 
     ManagedHsm = cmd.get_models('ManagedHsm', resource_type=ResourceType.MGMT_KEYVAULT)
     ManagedHsmSku = cmd.get_models('ManagedHsmSku', resource_type=ResourceType.MGMT_KEYVAULT)
 
     # tenantId and sku shouldn't be required
     profile = Profile(cli_ctx=cmd.cli_ctx)
-    _, _, tenant_id = profile.get_login_credentials(
-        resource=cmd.cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
```

Comment:
This is indeed why a token for AD Graph is retrieved.

https://github.com/Azure/msrestazure-for-python/blob/8849f398b6ebd4607de63c2f5d1318f44ec1d822/msrestazure/azure_active_directory.py#L592

```py
class MSIAuthentication(BasicTokenAuthentication):
    def __init__(self, port=50342, **kwargs):
        ....
        self.set_token()
```

When `msrestazure.azure_active_directory.MSIAuthentication` is initialized, it gets an access token immediately, not until `signed_session` or `get_token` is called, so `get_login_credentials` will trigger a web request to get AD Graph token.

Reply:
Another question, why does the original code choses `active_directory_graph_resource_id` instead of `active_directory_resource_id` (ARM's resource ID)? Is it because the name is confusing?

Reply:
> When `msrestazure.azure_active_directory.MSIAuthentication` is initialized, it gets an access token immediately, not until `signed_session` or `get_token` is called, so `get_login_credentials` will trigger a web request to get AD Graph token.

This explains why AAD call shows up when we create keyvault in Cloudshell sandbox as it's following MSI workflow.

Reply:
Anyway the whole call for `get_login_credentials` has been removed. Doesn't matter which resource id we need to use anymore😉